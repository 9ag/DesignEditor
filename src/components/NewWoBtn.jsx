import React from 'react'
class NewWoBtn extends React.Component {
  render() {
    const { onClick } = this.props
    return (
      <button onClick={onClick} type="button" className="float-add-btn" id="newWO">
        <i className="mdi-content-add fa-lg"></i>
      </button>
    )
  }
}

export default NewWoBtn