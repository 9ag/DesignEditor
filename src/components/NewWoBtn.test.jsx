
import { shallow, mount, render } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

import NewWoBtn from './NewWoBtn';
describe("NewWoBtn component test", () => {

    it("click callback test", () => {
      const callback = jest.fn();
      const component = shallow(<NewWoBtn onClick={callback}/>);
      component.simulate('click');
      expect(callback).toHaveBeenCalledTimes(1);
      component.simulate('click');
      expect(callback).toHaveBeenCalledTimes(2);
    });
     
    it('snapshot test', () => {
      const callback = jest.fn();
      const component = renderer.create(<NewWoBtn onClick={callback}/>).toJSON();
      expect(component).toMatchSnapshot();
    });

  
});