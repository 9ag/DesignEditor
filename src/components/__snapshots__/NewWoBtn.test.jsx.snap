// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewWoBtn component test snapshot test 1`] = `
<button
  className="float-add-btn"
  id="newWO"
  onClick={[MockFunction]}
  type="button"
>
  <i
    className="mdi-content-add fa-lg"
  />
</button>
`;
